### PR TITLE
feat: Add paraphase

### PR DESCRIPTION
**Pull request by MagdalenaZZ**
### This PR:

(If this is a release PR, no need to add following. Leave this part empty)
(Use the following lines to create a PR text body. Make sure to remove all non-relevant one after you're done)
(Repeat each field as many times as necessary)

Added: for new features.
Changed: for changes in existing functionality.
Deprecated: for soon-to-be removed features.
Removed: for now removed features.
Fixed: for any bug fixes.
Security: in case of vulnerabilities.

### Review and tests:
- [ ] Tests pass
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Code review
- [ ] `CHANGELOG.md` is updated
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

File: workflow/rules/paraphase.smk [hunk 1]
```diff
@@ -0,0 +1,49 @@
+__copyright__ = "Copyright 2023, Uppsala Universitet"
+__email__ = "magdalena.z@scilifelab.uu.se"
+__license__ = "GPL-3"
+
+# compile_paraphase_file_list(wildcards)
```
Comment (elleira) [inline workflow/rules/paraphase.smk:5]: Remove unused arguments (those commented)

File: workflow/rules/paraphase.smk [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        time=config.get("paraphase", {}).get("time", config["default_resources"]["time"]),
+    container:
+        config.get("paraphase", {}).get("container", config["default_container"])
+    message:
+        "{rule}: Calls SNVs on {input.bam} with paraphase to resolve SNVs in gene families"
+    wrapper:
+        "my-new-snakemake-wrapper-paraphrase/bio/paraphase"
+
+
+# "file:///beegfs-storage/projects/wp3/nobackup/Workspace/magz_testing/snakemake-wrappers/bio/paraphase"
```
Comment (elleira) [inline workflow/rules/paraphase.smk:47]: Should not be in hydra since only relevant for Uppsala

File: .tests/integration/config_pacbio.yaml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  fai: "reference/HES45.fna.fai"
+  design_bed: "reference/twist_DNA_solid.HES45.annotated.bed"
+
+paraphase:
+  genome: "--genome 38"
+  container: "docker://hydragenetics/paraphase:3.0.0"
+  fasta: "reference/chr16s.fa"
+  fai: "reference/homo_sapiens.fa.fai"
+  outfolder: "snv_indels/paraphase/"
+  # extra: "" # runs all appr 200 genes
```
Comment (elleira) [inline .tests/integration/config_pacbio.yaml:25]: Remove unused extras

File: .github/workflows/integration.yaml [hunk 4]
```diff
@@ -40,4 +40,5 @@ jobs:
         working-directory: .tests/integration
         run: |
           mamba install -c conda-forge -c bioconda singularity=3.8.6
-          snakemake -s ../../workflow/Snakefile -j 1 --show-failed-logs  --configfile config.yaml --use-singularity --singularity-args  " --cleanenv --bind /home/runner "
+          snakemake -s ../../workflow/Snakefile -j 1 --show-failed-logs --configfile config.yaml --use-singularity --singularity-args "--cleanenv --bind /home/runner"
+          snakemake -s ../../workflow/Snakefile -j 2 --show-failed-logs --configfiles config_pacbio.yaml --use-singularity --singularity-args "-e --cleanenv"
```
Comment (padraicc) [inline .github/workflows/integration.yaml:44]: This integration test is failing due to missing input files for paraphase.

**Review by padraicc — CHANGES_REQUESTED** [review]

File: workflow/rules/common.smk [hunk 5]
```diff
@@ -153,7 +153,21 @@ def get_glnexus_input(wildcards, input):
 
 
 def compile_output_list(wildcards: snakemake.io.Wildcards):
```
[the remark this answers is not in the document]
Comment (MagdalenaZZ) [reply workflow/rules/common.smk:155]: nu finns dom där

File: .github/workflows/integration.yaml (hunk 4 above)
Comment (MagdalenaZZ) [reply .github/workflows/integration.yaml:44]: They have been added, but paraphase will not work as intended before the PR https://github.com/snakemake/snakemake-wrappers/pull/3071 is accepted

File: workflow/rules/paraphase.smk [hunk 6]
```diff
@@ -0,0 +1,49 @@
+__copyright__ = "Copyright 2023, Uppsala Universitet"
+__email__ = "magdalena.z@scilifelab.uu.se"
+__license__ = "GPL-3"
+
+# compile_paraphase_file_list(wildcards)
+GENE = ["smn1", "CR1", "AMY1A", "CTAG1A", "BOLA2"]
```
[the remark this answers is not in the document]
Comment (MagdalenaZZ) [reply workflow/rules/paraphase.smk:6]: removed

**Pull request closed without merging**